UI/src/view/AssetLabel.js

```diff
@@ -33,23 +35,69 @@
         };
 
         this.renderFeatures = function(assets, zoomLevel, getPoint){
-            if(!me.isVisibleZoom(zoomLevel))
-                return [];
-
-            return _.chain(assets).
-                map(function(asset){
-                    var assetValue = me.getValue(asset);
-                    if(assetValue !== undefined){
-                        var style = me.getStyle(assetValue);
-                        var feature = me.createFeature(getPoint(asset));
-                        feature.setStyle(style);
-                        return feature;
-                    }
-                }).
-                filter(function(feature){ return feature !== undefined; }).
-                value();
+          if(!me.isVisibleZoom(zoomLevel))
+            return [];
+
+          return _.chain(assets).
+          map(function(asset){
+            var assetValue = me.getValue(asset);
+            if(assetValue !== undefined){
+              var style = me.getStyle(assetValue);
+              var feature = me.createFeature(getPoint(asset));
+              feature.setProperties(asset);
+              feature.setStyle(style);
+              return feature;
+            }
+          }).
+          filter(function(feature){ return feature !== undefined; })
+            .value();
         };
 
+      this.getCoordinateForGrouping = function (point){
+        var assetCoordinate = {lon : point.lon, lat : point.lat};
+        var assetCounter = {counter: 1};
+        if(_.isEmpty(me.populatedPoints)){
+          me.populatedPoints.push({coordinate: assetCoordinate, counter: 1});
+        }else{
+          var populatedPoint = _.find(me.populatedPoints, function (p) {
+            return me.isInProximity(point, p, me.MIN_DISTANCE);
+          });
+          if (!_.isUndefined(populatedPoint)) {
+            assetCoordinate = populatedPoint.coordinate;
+            assetCounter.counter = populatedPoint.counter + 1;
+            populatedPoint.counter++;
+          } else {
+            me.populatedPoints.push({coordinate: assetCoordinate, counter: 1});
+          }
+        }
+        return [[assetCoordinate.lon, assetCoordinate.lat], assetCounter.counter];
+      };
+
+      this.renderGroupedFeatures = function(assets, zoomLevel, getPoint){
+        if(!this.isVisibleZoom(zoomLevel))
+          return [];
+
+        return _.chain(assets).
+        map(function(asset){
+          var value = me.getValue(asset);
+          var assetLocation = getPoint(asset);
+          if(value !== undefined){
+            var styles = [];
```

Comment:
Why it's being used an empty array to save the styles? Isn't it always have just one position?